operator/src/main/java/dev/responsive/k8s/operator/reconciler/ManagedApplication.java

```diff
@@ -0,0 +1,246 @@
+package dev.responsive.k8s.operator.reconciler;
+
+import dev.responsive.k8s.crd.ResponsivePolicy;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.StatefulSet;
+import io.fabric8.kubernetes.client.dsl.AppsAPIGroupDSL;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ManagedApplication {
+
+  private final HasMetadata application;
+  private final Class<?> appClass;
+  private final String appName;
+  private final String namespace;
+
+  public ManagedApplication(HasMetadata application, Class<?> appClass,
+                            ResponsivePolicy policy) {
+    this (application, appClass, policy.getSpec().getApplicationName(),
+        policy.getSpec().getApplicationNamespace());
+  }
+
+  public ManagedApplication(HasMetadata application, Class<?> appClass,
+                            String appName, String namespace) {
+    this.application = application;
+    this.appClass = appClass;
+    this.appName = appName;
+    this.namespace = namespace;
+  }
+
+  public int getReplicas() {
+    if (appClass == Deployment.class) {
```

Comment:
Similarly, it's probably better to use an interface (or abstract base class) like:

```
interface ManagedApplication {
    int getReplicas();
    void setReplicas(int target, Context<ResponsivePolicy> context);
    String appType();
    String getResourceVersion();
}

class ManagedStatefulSet implements ManagedApplication {...}
class ManagedDeployment implements ManagedApplication {...}
```

Reply:
Feel free to address the second point in a follow-up if it makes sense to you.

Reply:
Agree on using an interface. Don’t know why I didn’t think of that. 

Will address it in a followup so that we have one build with statefulset support before Monday.